### PR TITLE
CRM: Automations Quote condition for status changed

### PR DESCRIPTION
**Pull request by gogdzl**
## Proposed changes:

* This PR adds the quote condition 'Status Changed' and respective tests.
* This PR also introduces a new function `jpcrm_get_quote_statuses` that returns quote statuses strings.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
https://github.com/Automattic/zero-bs-crm/issues/3192

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:

* Run tests by navigating to the CRM directory in your local installation after running `jetpack build plugins/crm`, and then run `composer phpunit -- --testdox --testsuite automation` (note this will currently run all tests as well with a new commit adding tests to this PR).

### CODE REVIEW

File: projects/plugins/crm/src/automation/commons/conditions/quotes/class-quote-status-changed.php [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+			case 'is_not':
+				$this->condition_met = ( $data['data'][ $field ] !== $value );
+				$this->logger->log( 'Condition met?: ' . ( $this->condition_met ? 'true' : 'false' ) );
+
+				return;
+		}
+		$this->condition_met = false;
+		$this->logger->log( 'Invalid operator: ' . $operator );
+
+		throw new Automation_Exception( 'Invalid operator: ' . $operator );
```
Comment (robertf4) [inline projects/plugins/crm/src/automation/commons/conditions/quotes/class-quote-status-changed.php]: I have the same suggestion here as the previous PR I reviewed: check this at the beginning of the function using `$valid_operators` and then have a default case in the switch that throws an "unimplemented operator exception".

**Review by cleacos — CHANGES_REQUESTED** [review]
The quote model doesn't have the `status` property. Determining the virtual status of a quote requires performing some checks with the accepted field.

**Comment by gogdzl** [comment]
> The quote model doesn't have the status property. Determining the virtual status of a quote requires performing some checks with the accepted field.

Thanks a lot for pointing that out! Sometimes quotes do have a status field, but just in some cases ( e.g. sorting ), and they are indeed calculated in the code.

I added a fix for that and everything should be corrected.

File: projects/plugins/crm/src/automation/commons/conditions/quotes/class-quote-status-changed.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+	/**
+	 * Get the category of the quote status changed condition.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return string The category 'jpcrm/quote_condition'.
+	 */
+	public static function get_category(): string {
+		return 'quote';
```
Comment (cleacos) [inline projects/plugins/crm/src/automation/commons/conditions/quotes/class-quote-status-changed.php]:
```suggestion
		return __( 'quote', 'zero-bs-crm' );
```

**Review by cleacos — DISMISSED** [review]
Cool. The code looks good, and the tests passed. I committed a small change (category translation)

LGTM!

**Review by robertf4 — APPROVED** [review]
I approve this PR, it is the best!

**Pull request merged**